internal/rpcapi/dynrpcserver/generator/main.go

```diff
@@ -94,6 +94,16 @@ func main() {
 			filename := toFilenameCase(baseName) + ".go"
 			absFilename := filepath.Join(outDir, filename)
 
+			if regexp.MustCompile("^Unsafe").MatchString(ifaceName) {
+				// This isn't a gRPC server interface, so skip it.
+				//
+				// This is an interface that's intended to be embedded to help users to meet requirements for Unimplemented servers.
+				// See:
+				// > Docs: https://pkg.go.dev/google.golang.org/grpc/cmd/protoc-gen-go-grpc#readme-future-proofing-services
+				// > PR for Unsafe interfaces: https://github.com/grpc/grpc-go/pull/3911
+				continue Types
+			}
```

Comment:
This change is related to [this issue I highlighted in an earlier comment](https://github.com/hashicorp/terraform/pull/37685#issuecomment-3345958140). This script is written expecting .pb.go files generated from the old `github.com/golang/protobuf/protoc-gen-go` tooling. With the new tooling in this PR there are some new interfaces that this script needs to be updated to tolerate.

I took inspiration from L83-L85, but I need review from someone more familiar with the purpose of this script (paging @liamcervante).